src/fvectorbin/fvector-add-noise.cc

```diff
@@ -0,0 +1,53 @@
+#include "base/kaldi-common.h"
+#include "util/common-utils.h"
+#include "fvector/fvector-perturb.h"
+
+int main(int argc, char *argv[]) {
+  try {
+    using namespace kaldi;
+    const char *usage =
+        "Perturb the chunk data.\n"
+        "Usage:  fvector-add-noise [options...] <chunk-rspecifier> <perturbed-wspecifier>\n";
+
+    // construct all the global objects
+    ParseOptions po(usage);
+    FvectorPerturbOptions perturb_opts;
+    perturb_opts.Register(&po);
+
+    po.Read(argc, argv);
+    if (po.NumArgs() != 2) {
+      po.PrintUsage();
+      exit(1);
+    }
+
+    std::string chunk_rspecifier = po.GetArg(1),
+      perturbed_chunk_rspecifier = po.GetArg(2);
+
+    SequentialBaseFloatMatrixReader chunk_reader(chunk_rspecifier);
+    BaseFloatMatrixWriter perturbed_chunk_writer(perturbed_chunk_rspecifier);
+
+    int64 num_read = 0, num_written = 0;
+    for (; !chunk_reader.Done(); chunk_reader.Next(), num_read++) {
+      std::string key = chunk_reader.Key();
+      // input_chunk has 3 lines.
+      const Matrix<BaseFloat> &input_chunk = chunk_reader.Value();
+      // whole_chunk has 4 lines, it copies the first line and will be operate.
+      Matrix<BaseFloat> whole_chunk(4, input_chunk.NumCols());
+      MatrixIndexT indices[4] = {0, 0, 1, 2};
+      whole_chunk.CopyRows(input_chunk, indices);
```

Comment:
I am not sure, what you are doing here. Would you describe it in more details?

Reply:
In "input_chunk" matrix, it contains 3 lines(S1, N1, N2). Here, we copy the S1. So, in "whole_chunk" matrix, it contains 4 lines(S1, S2, N1, N2). When we do "add-noise", we will add additive noise(N1->S1 and N2->S2). So after "add-noise" operation, we has a matrix which contains two rows. And the two rows(100ms) come from the same source(120ms).

Reply:
What is the number of rows in  your input chunk? is this 3? If it is 3, I don't think it is necessary to have this implementation and you can have noise_mat and orig_mat and the output is perturbed version of each row in orig_mat with rows in noise_mat. 

Also you mean 'adding another speech chunk' by noise? 
Please describe it with detail in code!